### ISSUE TITLE
"deps/googletest does not contain a CMakeLists.txt file" when building from source tarball

### ISSUE THREAD

**Issue by HighCommander4** [cpp-netlib/cpp-netlib#813]
I am trying to build cpp-netlib 0.13.0-rc1 from the source tarball located [here](http://downloads.cpp-netlib.org/0.13.0/cpp-netlib-0.13.0-rc1.tar.bz2).

When invoking cmake, I get:

```
CMake Error at CMakeLists.txt:128 (add_subdirectory):
  The source directory

    /home/botond/dev/libraries/cpp-netlib-0.13.0-rc1/deps/googletest

  does not contain a CMakeLists.txt file.
```

I understand that when obtaining the source from git, this dependency is obtained via submodules, but there should be a documented way to obtain it when building from a source tarball as well.

**Comment by deanberris** [comment]
Correct. I think the CMake configuration is due for an overhaul, to check whether google test is already installed in the system and using that instead. Some of the dependencies at build time could just be removed from the submodules, and are optional anyway (in this case google test).

**Comment by Synoecium** [comment]
one year has passed and nothing changed. Is this lib supported at all or outdated?

**Comment by deanberris** [comment]
@Synoecium It has not changed -- we just haven't found a good solution to this in the packaged files. The options are to "vendor" googletest (i.e. will be included in the download) or to remove the explicit dependency. If you want to use the release version via git (don't need to use the prepackaged files) then see the 0.13-release branch with the submodules.

How critical is it that you build the tests too? There's an option in CMakeLists.txt named `CPP-NETLIB_BUILD_TESTS` which is default-enabled. You can set this to `OFF` when building with cmake to work around the problem.

**Comment by Synoecium** [comment]
thank you for answer. No, it isn't critical to build the tests. In my case i just took official release packages, it's ok for me.

**Comment by emerth** [comment]
Using Git cloned repo, Sept 6 2019, Visual Studio 15 (2017) and Windows 10.

I've changed the CMakeLists.txt to set CPP-NETLIB_BUILD_TESTS to OFF.

I'm running cmake thus:

> C:\cpp-netlib-0.13.0\cpp-netlib-0.13.0-build>cmake.exe -G "Visual Studio 15 2017 Win64" -DBOOST_ROOT:string="C:\stories\powerbi\cpp-netlib-0.13.0\boost_1_68_0" -DCPP-NETLIB_BUILD_TESTS:string="OFF" ..\cpp-netlib

Yet it still fails with message:

> CMake Error at deps/uri/CMakeLists.txt:85 (add_subdirectory):
>   The source directory
> 
>     C:/Stories/PowerBI/cpp-netlib-0.13.0/cpp-netlib/deps/uri/deps/googletest
> 
>   does not contain a CMakeLists.txt file.
> 

Is it possible to build this on Windows at all?
